workloads/network-perf/common.sh

```diff
@@ -43,11 +44,41 @@ export_defaults() {
   export pin=true
   export networkpolicy=${NETWORK_POLICY:=false}
   export multi_az=${MULTI_AZ:=true}
+  export baremetalCheck=$(oc get infrastructure cluster -o json | jq .spec.platformSpec.type)
   zones=($(oc get nodes -l node-role.kubernetes.io/workload!=,node-role.kubernetes.io/infra!=,node-role.kubernetes.io/worker -o go-template='{{ range .items }}{{ index .metadata.labels "topology.kubernetes.io/zone" }}{{ "\n" }}{{ end }}' | uniq))
   platform=$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.type}' | tr '[:upper:]' '[:lower:]')
   log "Platform is found to be : ${platform} "
-  # If multi_az we use one node from the two first AZs
-  if [[ ${platform} == "vsphere" ]]; then
+
+  #Check to see if the infrastructure type is baremetal to adjust script as necessary 
+  if [[ "${baremetalCheck}" == '"BareMetal"' ]]; then
+    log "BareMetal infastructure: setting isBareMetal accordingly"
+    export isBareMetal=true
+  else
+    export isBareMetal=false
+  fi
+
+  #If using baremetal we use different query to find worker nodes
+  if [[ "${isBareMetal}" == "true" ]]; then
+    log "Colocating uperf pods for baremetal"
```

Comment:
We are not really colocating are we?

Reply:
right, we are not. we randomly pick 2 worker nodes. I will delete this log.